data/provider/src/main/java/tech/pegasys/teku/api/peer/Eth2PeerWithEnr.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Consensys Software Inc., 2024
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.api.peer;
+
+import java.util.Optional;
+import tech.pegasys.teku.networking.eth2.peers.Eth2Peer;
+
+public record Eth2PeerWithEnr(Eth2Peer peer, Optional<String> enr) {}
```

Comment:
As the `enr` is optional, does it make sense to just add this field to `EthPeer`?